lib/backend/k8s/syncer.go

```diff
@@ -424,18 +424,20 @@ func (syn *kubeSyncer) performSnapshot() ([]model.KVPair, map[string]bool, resou
 
 		versions.podVersion = poList.ListMeta.ResourceVersion
 		for _, po := range poList.Items {
-			// Ignore any updates for host networked pods.
-			if syn.kc.converter.isHostNetworked(&po) {
-				log.Debugf("Skipping host networked pod %s/%s", po.ObjectMeta.Namespace, po.ObjectMeta.Name)
+			// Ignore any updates for pods which are not ready / valid.
+			if !syn.kc.converter.isCalicoPod(&po) {
+				log.Debugf("Skipping pod %s/%s", po.ObjectMeta.Namespace, po.ObjectMeta.Name)
 				continue
 			}
 
 			// Convert to a workload endpoint.
-			wep, _ := syn.kc.converter.podToWorkloadEndpoint(&po)
-			if wep != nil {
-				snap = append(snap, *wep)
-				keys[wep.Key.String()] = true
+			wep, err := syn.kc.converter.podToWorkloadEndpoint(&po)
+			if err == nil {
```

Comment:
@robbrockbank I've added some FV infrastructure to this PR which validates that updates actually do show up in the syncer.

It doesn't go all the way, in that:
- I haven't added it to all of the different resources tests yet.
- It doesn't check the content yet, just that an update was received.

It catches the above `==` vs `!=` bug.  I think this is good enough for now, but we should fix those two bullets in the near future to make these tests more robust.